### DISCUSSION TITLE
Chart lines colors/width/style and/or title/subtitle customization

### DISCUSSION THREAD

**Discussion by andreltr** [serkor1/cryptoQuotes#19] [Q&A] [answered]
Hello,

Is there a way to customize the lines colors/width/style of a chart?  I tried color = "red" and it did not work.
How about adding a title and/or subtitle to it?

TIA.

**Comment by serkor1** [comment, marked as answer]
Hi @andreltr,

The `chart()`-function and family is rather high-level and restricted in customization. However, some elements of the chart can be modified in its `color`. Below is an example for the `bollinger_bands()`-function,

``` r
library(cryptoQuotes)
```

``` r
# get BTCUSDT
# pair
BTC <- get_quote(
  ticker = "BTCUSDT",
  source = "bybit"
)
```

``` r
# chart with bollinger bands
# in red
chart(
  ticker = BTC,
  main   = kline(),
  sub    = list(
    volume()
  ),
  indicator = list(
    bollinger_bands(
      color = "red"
    )
  )
)
```

![](https://i.imgur.com/RYPbEH7.png)

### Hacking the `chart()`-function

The rendered chart is from {plotly}, and can be modified using the `layout()`-function. Below is an example where the `title` is changed and a `subtitle` is changed,

``` r
library(cryptoQuotes)
```

``` r
# get BTCUSDT
# pair
BTC <- get_quote(
  ticker = "BTCUSDT",
  source = "bybit"
)
```

``` r
# chart with bollinger bands
# and custom titles and subtitle
plotly::layout(
  p = chart(
    ticker = BTC,
    main   = kline(),
    sub    = list(
      volume()
    ),
    indicator = list(
      bollinger_bands()
    )
  ),
  title = list(
    text = "Custom Title <br> Subtitle"
  )
)
```

![](https://i.imgur.com/KNDx8x1.png)


### Charting with {quantmod}

Alternatively, you can use {quantmod} for charting instead of {cryptoQuotes}; the `chartSeries()`-function is far superior when it comes to customization of charts



<sup>Created on 2024-06-23 with [reprex v2.1.0](https://reprex.tidyverse.org)</sup>
